core/domain/acl_decorators.py

```diff
@@ -0,0 +1,121 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Decorators to provide authorization across the site."""
+
+from core.domain import rights_manager
+from core.domain import role_services
+import feconf
+
+
+def check_activity_accessible(self, activity_id, activity_type):
```

Comment:
Nit: activity_type should come before activity_id.

Also, 'self' probably should be named something else. Maybe 'handler' or 'handler_instance' -- depending on what it is? Even better -- remove 'self' altogether and pass in the various params (actions, user_id) explicitly.